Update somfy_mylink Cover with an explicit default for _attr_is_closed

cthornton opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
The base `Cover` entity requires an explicit value for `_attr_is_closed`.

Since the `SomfyShade` is an assumed state, we don't know by default whether the shade is open or not, so we need to
explicitly return `None` for `_attr_is_closed`



## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #65690
- This PR is related to issue: n/a
- Link to documentation pull request:  n/a

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [x] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

homeassistant commented:
Hi @cthornton, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#68053).

Once you do that we will be able to review and accept this pull request.

Thanks!

elupus commented on homeassistant/components/somfy_mylink/cover.py:
```diff
@@ -65,6 +65,9 @@ class SomfyShade(RestoreEntity, CoverEntity):
     _attr_should_poll = False
     _attr_assumed_state = True
 
+    # Assumed state, by default we don't know if the shade is closed or not
```
Please remove the comment. It is mostly helpfull now, so fits better in the commit message.

elupus commented on tests/components/somfy_mylink/test_somfy_shade.py:
```diff
@@ -0,0 +1,16 @@
+    assert somfy_shade.state is None
+    assert somfy_shade.is_closed is None
+    assert somfy_shade.is_closing is None
+    assert somfy_shade.is_opening is None
```
While it is good to add tests, we want tests for integrations to test with states on state machine, not test internal objects. So the integration should be set up and the resulting hass.states values should be checked.

cthornton replied:
I don't think the existing tests have any coverage of the states of the shades -- should I remove this test, or would you prefer I figure out how to get test coverage of this?

elupus replied:
Remove it from this pull for now. We would very much appreciate tests so if you can work on that separate it would be much appreciated.

cthornton replied:
Thanks! I made sure to manually test this on a local hass server on my machine, and it seems to work

elupus commented on homeassistant/components/somfy_mylink/cover.py:
```diff
@@ -64,6 +64,7 @@ class SomfyShade(RestoreEntity, CoverEntity):
 
     _attr_should_poll = False
     _attr_assumed_state = True
+    _attr_is_closed = None
```
I missed that this should not be a class constant. Set it in init since it is updated by the state updates.

cthornton committed:
Default somfy_mylink shade's _attr_is_closed to `None`

The base `Cover` entity requires an explicit value for
`_attr_is_closed`.

Since the `SomfyShade` is an assumed state, we don't know
by default whether the shade is open or not, so we need to
explicitly return `None` for `_attr_is_closed`
